pkg/metadataapi/server/api/server.go

```diff
@@ -35,6 +35,10 @@ func (s *Server) Route(r *mux.Router) {
 	r.HandleFunc("/environment", s.GetEnvironment).Methods(http.MethodGet)
 	r.HandleFunc("/environment/{name}", s.GetEnvironmentVariable).Methods(http.MethodGet)
 
+	// Logs
+	r.HandleFunc("/logs", s.PostLog).Methods(http.MethodPost)
+	r.HandleFunc("/logs/messages", s.PostLogMessage).Methods(http.MethodPost)
```

Comment:
Wouldn't we want this to be `/logs/{logId}/messages`?

Reply:
I don't believe so. The log ID is embedded in the log object we're sending and is a fairly critical part of it. Adding the {logId} here feels like we're trying to maintain a REST API structure just for the sake of it. I don't like situations where we have the same data duplicated in two different places for no reason. That directly leads to (at least) two fundamental software engineering problems.

I believe this is a closer representation to the streaming nature here, and I don't see a need to really separate this out artificially. Everything about it feels redundant.

Originally the API signature included the {logId} and used it's own log structure that did not include the log ID and only the message. And I think that makes sense if you want to include the {logId} in the API. But if we're going to reuse the PLS structure (which I think is a better direction) and I believe what we called for in the spec, it makes more sense to have these APIs represent the gRPC functionality a bit closer.

I don't really think it matters either way, and I have a hard time justifying adding the {logId} in for no real reason.

Reply:
This is another instance where the benefits to using gRPC would be somewhat nullified (i.e. in this case, using a resource-based system anyway). I don't believe this feels like a "resource" per se. Obviously it could be viewed either way, but a "log" in this case does not feel like a resource. The log id feels like what it is: an index to identify the log.